### DISCUSSION TITLE
configure: error: Unexpected output of 'arch' on OSX (on Mac while installing python 3.8.8)

### DISCUSSION THREAD

**Discussion by VijayN10** [pyenv/pyenv#2907] [Q&A] [answered]
Hello,

I am trying to install python 3.8.8 on macOS (M1 Sonoma Version 14.3.1), I am getting the following message related to zlib. 

```
q21422vn@E-LOSXG891WFV ~ % pyenv --version
pyenv 2.3.36
q21422vn@E-LOSXG891WFV ~ % pyenv install 3.8.8
python-build: use openssl from homebrew
python-build: use readline from homebrew
Downloading Python-3.8.8.tar.xz...
-> https://www.python.org/ftp/python/3.8.8/Python-3.8.8.tar.xz
Installing Python-3.8.8...
patching file 'Misc/NEWS.d/next/Build/2021-10-11-16-27-38.bpo-45405.iSfdW5.rst'
patching file configure
patching file configure.ac
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 14.3.1 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/q4/4btz2xfd5y72jv6gp5lzx_680000gp/T/python-build.20240221175322.82175
Results logged to /var/folders/q4/4btz2xfd5y72jv6gp5lzx_680000gp/T/python-build.20240221175322.82175.log

Last 10 log lines:
checking size of _Bool... 1
checking size of off_t... 8
checking whether to enable large file support... no
checking size of time_t... 8
checking for pthread_t... yes
checking size of pthread_t... 8
checking size of pthread_key_t... 8
checking whether pthread_key_t is compatible with int... no
configure: error: Unexpected output of 'arch' on OSX
make: *** No targets specified and no makefile found.  Stop.
q21422vn@E-LOSXG891WFV ~ % 

```

Could anyone please help. 

Thanks!

**Comment by native-api** [comment]
You likely have Homebrew binutils on PATH.
https://github.com/pyenv/pyenv/wiki/Common-build-problems#keg-only-homebrew-packages-are-forcibly-linked--added-to-path

**Reply by VijayN10** [reply]
Thank you for your reply. 

I tried unlinking `binutils` `llvml` `lcoreutils` using `brew unlink` command. 
and got following - 

```
q21422vn@E-LOSXG891WFV ~ % brew unlink binutils
Error: No such keg: /opt/homebrew/Cellar/binutils

q21422vn@E-LOSXG891WFV ~ % brew unlink llvm
brew unlink coreutils

Error: No such keg: /opt/homebrew/Cellar/llvm
Unlinking /opt/homebrew/Cellar/coreutils/9.4... 256 symlinks removed.
q21422vn@E-LOSXG891WFV ~ % 

```

Which means that `bunutils `(which is reponsible for my error) is not linked to path. 
However, after unlinking `coreutils` (which seems to be linked), I tried again to install version 3.8.8 but got the same error. 

Interestingly I am able to install version 3.8.10

Could you please tell me how to proceed? 

Thanks.

**Reply by native-api** [reply]
https://github.com/pyenv/pyenv/wiki/Common-build-problems#configure-error-internal-configure-error-for-the-platform-triplet-please-file-a-bug-report-in-macos

3.8.8 doesn't support Arm64. Although the error was different when we discovered that.

**Reply by VijayN10** [reply]
I see. Then would you please suggest any work around to install 3.8.8 on Mac?

**Reply by native-api** [reply, marked as answer]
Install the latest micro release which does have compatibility patches. Micro releases are supposed to be compatible with each other.
